### PR TITLE
[python/zh-cn]Complete Translation

### PR DESCRIPTION
**Pull request by mapleincode**
- [x] I solemnly swear that this is all original content of which I am the original author
- [x] Pull request title is prepended with `[language/lang-code]` (example `[python/fr-fr]` or `[java/en]`)
- [x] Pull request touches only one file (or a set of logically related files with similar changes made)
- [x] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)
- [x] If you've changed any part of the YAML Frontmatter, make sure it is formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
  - [x] Yes, I have double-checked quotes and field names!

### CODE REVIEW

File: zh-cn/python-cn.html.markdown [hunk 1]
```diff
@@ -512,53 +735,225 @@ class Human(object):
...
+from bat import Bat
+
+# 定义 Batman 作为子类，来同时继承 SuperHero 和 Bat
+class Batman(Superhero, Bat):
+
+    def __init__(self, *args, **kwargs):
+        # 通常要继承属性，你必须调用 super:
+        # super(Batman, self).__init__(*args, **kwargs)
+        # 然而在这里我们处理的是多重继承，而 super() 只会返回 MRO 列表的下一个基础类。
+        # 因此，我们需要显示的调用初始类的 __init___
```
Comment (imba-tjd) [inline zh-cn/python-cn.html.markdown:917]:
```suggestion
        # 因此，我们需要显式调用初始类的 __init__
```

File: zh-cn/python-cn.html.markdown [hunk 2]
```diff
@@ -470,39 +642,90 @@ add_10 = create_adder(10)
...
+# 我们使用 "class" 语句来创建类
+class Human:
+
+    # 一个类的字段。 这个字段共享给这个类的所有实例。
     species = "H. sapiens"
 
     # 构造方法，当实例被初始化时被调用。注意名字前后的双下划线，这是表明这个属
-    # 性或方法对Python有特殊意义，但是允许用户自行定义。你自己取名时不应该用这
-    # 种格式。
+    # 性或方法对 Python 有特殊意义，但是允许用户自行定义。
```
Comment (imba-tjd) [inline zh-cn/python-cn.html.markdown:708]: `属性`放到同一行里吧

File: zh-cn/python-cn.html.markdown [hunk 3]
```diff
@@ -145,6 +185,13 @@ bool(()) # => False
...
 
+# 默认情况下，print 函数会在输出结果后加入一个空行作为结尾
+# 可以使用附加参数改变输出结尾
+print("Hello, World", end="!")  # => Hello, World!
+
+# 可以很简单的从终端获得输入数据
+input_string_var = input("Enter some data: ") # 返回字符串数值
+
 # 在给变量赋值前不用提前声明
 # 传统的变量命名是小写，用下划线分隔单词
```
Comment (imba-tjd) [inline zh-cn/python-cn.html.markdown:196]:
```suggestion
# 习惯上变量命名是小写，用下划线分隔单词
```

File: zh-cn/python-cn.html.markdown [hunk 4]
```diff
@@ -35,50 +36,66 @@ Python 是由吉多·范罗苏姆(Guido Van Rossum)在 90 年代早期设计。
...
+
+# 数值与 True 和 False 之间的运算
 0 == False # => True
 2 == True # => False
 1 == True # => True
+-5 != False # => True
+
+# 使用布尔逻辑运算符对数字类型的值进行运算时，会把数值强制转换为布尔值进行运算
+# 但计算结果会返回它们的强制转换前的值
+# 注意不要把 bool(ints) 与位运算、and/or (&, |) 混淆
```
Comment (imba-tjd) [inline zh-cn/python-cn.html.markdown:93]:
```suggestion
# 注意不要把 bool(ints) 与位运算的 "按位与"、"按位或" (&, |) 混淆
```

File: zh-cn/python-cn.html.markdown [hunk 5]
```diff
@@ -35,50 +36,66 @@ Python 是由吉多·范罗苏姆(Guido Van Rossum)在 90 年代早期设计。
...
-5.0 // 3.0 # => 1.0 # 浮点数也可以
--5 // 3  # => -2
--5.0 // 3.0 # => -2.0
+5 // 3       # => 1
+5.0 // 3.0   # => 1.0 # 浮点数也可以
+-5 // 3      # => -2
+-5.0 // 3.0  # => -2.0
 
 # 浮点数的运算结果也是浮点数
 3 * 2.0 # => 6.0
```
Comment (imba-tjd) [inline zh-cn/python-cn.html.markdown:54]: 现在原文没有这句话了，有的是

```
# The result of division is always a float
10.0 / 3  # => 3.3333333333333335
```

File: zh-cn/python-cn.html.markdown [hunk 6]
```diff
@@ -35,50 +36,66 @@ Python 是由吉多·范罗苏姆(Guido Van Rossum)在 90 年代早期设计。
...
 
-# 整数也可以当作布尔值
-0 and 2 # => 0
--5 or 0 # => -5
+# True 和 False 实质上就是数字 1 和0
+True + True # => 2
+True * 8    # => 8
+False - 5   # => -5
+
+# 数值与 True 和 False 之间的运算
```
Comment (imba-tjd) [inline zh-cn/python-cn.html.markdown:85]:
```suggestion
# 数值与 True 和 False 之间的比较运算
```

File: zh-cn/python-cn.html.markdown [hunk 7]
```diff
@@ -154,7 +201,7 @@ some_var  # => 5
 # 参考流程控制一段来学习异常处理
 some_unknown_var  # 抛出NameError
```
Comment (imba-tjd) [inline zh-cn/python-cn.html.markdown:203]: 这里少了一段

```
# if can be used as an expression
# Equivalent of C's '?:' ternary operator
"yay!" if 0 > 1 else "nay!"  # => "nay!"

```

**Review by imba-tjd — CHANGES_REQUESTED** [review]
我还建议open()的部分提一下encoding='utf-8'。

另外，markdown的checkbox的语法，`- [ ]`是未勾选，`- [x]`是勾选。不是打√

File: zh-cn/python-cn.html.markdown (hunk 5 above)
Comment (mapleincode) [reply zh-cn/python-cn.html.markdown:54]: 原文是把 `35 / 5 # 7.0` 作为普通案例，我这边把除法和其他三个区分开来。
而且  `35 / 5 # 7.0`  比 `10.0 / 3` 更能提现 `The result of division is always a float`

Comment (mapleincode) [reply zh-cn/python-cn.html.markdown:54]: 而 `浮点数的运算结果也是浮点数` 这个是否也可以作为知识点保留？

Comment (imba-tjd) [reply zh-cn/python-cn.html.markdown:54]: 行吧

**Review by imba-tjd — APPROVED** [review]
I will wait for a week before merging in order to see whether someone else wants to review.

File: zh-cn/python-cn.html.markdown [hunk 8]
```diff
@@ -509,16 +509,19 @@ with open("myfile2.txt", "w+") as file:
...
 # print: {"aa": 12, "bb": 21}
 
-with open('myfile2.txt', "r+") as file:
+with open("myfile2.txt", "r+") as file:
     contents = json.load(file)       # 从文件读取 json 对象
 print(contents)
 # print: {"aa": 12, "bb": 21}
 
+with open("myfile3.txt", "r+", encoding = "utf-8") as file:
+    contents = file.read()           # 从文件以 utf-8 的编码读取字符串（默认为 ascii）
```
Comment (imba-tjd) [inline zh-cn/python-cn.html.markdown:523]: 默认不是ascii，是win下默认是所谓的*ANSI*，Linux下默认是u8。没必要加在这里，直接给512行后面加一下注释，说win下一般要指定encoding = "utf-8"等等就好了

**Comment by imba-tjd** [comment]
Thanks for your contribution!

**Pull request merged**